common/src/main/java/org/apache/seata/common/metadata/Node.java

```diff
@@ -96,19 +102,92 @@ public void setInternal(Endpoint internal) {
         this.internal = internal;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(control, transaction);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Node node = (Node) o;
+        return Objects.equals(control, node.control) && Objects.equals(transaction, node.transaction);
```

Comment:
if two nodes have the same ip and port,they will be considered as one node,even if their weight,role...etc are different with each other. Because their value can be chages with time.